### ISSUE TITLE
Permalink DuplicatePermalinkOutputError when using `.webc` with pagination in 2.0.0-canary.16

### ISSUE THREAD

**Issue by rijkvanzanten** [11ty/eleventy#2615]
**Describe the bug**

When rendering a paginated page that is a `.webc` component, I'm getting a DuplicatePermalinkOutputError. It seems to work as expected with `njk` or other rendering engines (from what I can tell). This feels very similar to https://github.com/11ty/eleventy/issues/2333, but seeing that was confirmed fixed, I'm not sure if it's the exact same cause.

**To Reproduce**

Setup the following structure:

```
src/
  _data/
    test.json
  index.webc
.eleventy.js
```

with the following contents:

**src/_data/test.json**

```json
[{ "title": "Test 1" }, { "title": "Test 2" }, { "title": "Test 3" }]
```

**src/index.webc**

```html
---
pagination:
  data: test
  size: 1
  alias: item
permalink: "thing/{{ item.title | slugify }}/"
---

<h1>{{ item.title }}</h1>
```

**.eleventy.js**

```js
const pluginWebc = require("@11ty/eleventy-plugin-webc");

module.exports = function (eleventyConfig) {
	eleventyConfig.addPlugin(pluginWebc, {
		components: "./src/_components/**/*.webc",
	});

	return {
		dir: {
			input: "src",
			output: "dist",
		},
	};
}
```

When you try building the site with eleventy, witness the following error:

```
> eleventy 

[11ty] Problem writing Eleventy templates: (more in DEBUG output)
[11ty] Output conflict: multiple input files are writing to `dist/thing/{{ item.title | slugify }}/index.html`. Use distinct `permalink` values to resolve this conflict.
[11ty]   1. ./src/index.webc
[11ty]   2. ./src/index.webc
[11ty]   3. ./src/index.webc (via DuplicatePermalinkOutputError)
[11ty]
[11ty] Original error stack trace: (Repeated output has been truncated…) [...truncated]
```

**Expected behavior**

By changing `.webc` to `.njk` or another rendering engine, note how it works as expected:

```
> eleventy

[11ty] Writing dist/thing/test-1/index.html from ./src/index.njk
[11ty] Writing dist/thing/test-2/index.html from ./src/index.njk
[11ty] Writing dist/thing/test-3/index.html from ./src/index.njk
[11ty] Wrote 3 files in 0.10 seconds (v2.0.0-canary.16)
```

WebC should work the same here

**Environment:**

- OS and Version: macOS 12.6, Node 18.11.0
- Eleventy Version: 2.0.0-canary.16
- Eleventy Plugin WebC version: 0.5.1

**Comment by rijkvanzanten** [comment]
FWIW the custom dir input/output configuration in `.eleventy.js` doesn't seem to matter for this bug

**Comment by rijkvanzanten** [comment]
Update!

This is caused by the fact that the `permalink` is rendered by the engine used for the current file (for example a `.liquid` file uses Liquid for the permalink). For most of the default supported rendering engines, the `{{ }}` syntax for the permalink (as per the docs) works to render a basic string inline, however `webc` doesn't have such a syntax. Instead, to make things work, you gotta use `permalink: 'books/<template webc:type="11ty" 11ty:type="njk">{{ book.title | slugify }}</template>/'` instead of `permalink: "books/{{ book.title | slugify }}/"`

**Comment by rijkvanzanten** [comment]
For other people coming across this, you can achieve the same _without_ having to rely on a nested rendering engine as follows:


```
permalink: '/example/<template webc:nokeep @html="slugify(data.example)"></template>/'
```

**Comment by zachleat** [comment]
Just for future visitors this bug was fixed with WebC v0.7.0! https://github.com/11ty/eleventy-plugin-webc/issues/27 No workaround needed!
